### ISSUE TITLE
Code Table Request - Mexican wolf studbook number

### ISSUE THREAD

**Issue by dustymc** [ArctosDB/arctos#5936]
## Instructions

This is a template to facilitate communication with the Arctos Code Table Committee. Submit a separate request for each relevant value. This form is appropriate for exploring how data may best be stored, for adding vocabulary, or for updating existing definitions.

Reviewing documentation before proceeding will result in a more enjoyable experience.

* [Issue Documentation](http://handbook.arctosdb.org/how_to/How-to-Use-Issues-in-Arctos.html)
* [Code Table Documentation](https://handbook.arctosdb.org/how_to/How-To-Manage-Code-Table-Requests.html)
* [Video Tutorial - Submit a Code Table Request](https://youtu.be/t2jHbsRA3lk)

------------------------------

## Initial Request

**Goal**: Describe what you're trying to accomplish. This is the only necessary step to start this process. The Committee is available to assist with all other steps. Please clearly indicate any uncertainty or desired guidance if you proceed beyond this step.

All [Mexican wolf studbook number](https://arctos.database.museum/info/ctDocumentation.cfm?table=ctcoll_other_id_type#mexican_wolf_studbook_number) should be replaced with other ID type = studbook and issued by agent [United States Fish and Wildlife Service Mexican Wolf Recovery Program](https://arctos.database.museum/agent/1012059)

**Proposed Value**: Proposed new value. This should be clear and compatible with similar values in the relevant table and across Arctos. 

**Proposed Definition**: Clear, complete, non-collection-type-specific **functional** definition of the value. Avoid discipline-specific terminology if possible, include parenthetically if unavoidable.

**Context**: Describe why this new value is necessary and existing values are not.

**Table**: Code Tables are http://arctos.database.museum/info/ctDocumentation.cfm. Link to the specific table or value. This may involve multiple tables and will control datatype for Attributes. OtherID requests require BaseURL (and example) or explanation. Please ask for assistance if unsure.

**Collection type**: Some code tables contain collection-type-specific values. ``collection_cde`` may be found from https://arctos.database.museum/home.cfm

**Priority**: Please describe the urgency and/or choose a priority-label to the right. You should expect a response within two working days, and may utilize [Arctos Contacts](https://arctosdb.org/contacts/) if you feel response is lacking.

**Available for Public View**: Most data are by default publicly available. Describe any necessary access restrictions. 

**Project**: Add the issue to the [Code Table Management Project](https://github.com/ArctosDB/arctos/projects/13#card-31628184).

**Discussion**: Please reach out to anyone who might be affected by this change. Leave a comment or add this to the Committee agenda if you believe more focused conversation is necessary.

## Approval

All of the following must be checked before this may proceed.

The [How-To Document](https://handbook.arctosdb.org/how_to/How-To-Manage-Code-Table-Requests.html) should be followed. Pay particular attention to terminology (with emphasis on consistency) and documentation (with emphasis on functionality).

- [x] Code Table Administrator[1] - check and initial, comment, or thumbs-up to indicate that the request complies with the how-to documentation and has your approval
- [ X] Code Table Administrator[2] - check and initial, comment, or thumbs-up to indicate that the request complies with the how-to documentation and has your approval
- [x] DBA - The request is functionally acceptable. The term is not a functional duplicate, and is compatible with existing data and code.
- [x] DBA - Appropriate code or handlers are in place as necessary. (ID_References, Media Relationships, Encumbrances, etc. require particular attention)

## Rejection

If you believe this request should not proceed, explain why here. Suggest any changes that would make the change acceptable, alternate (usually existing) paths to the same goals, etc. 

1. Can a suitable solution be found here? If not, proceed to (2)
2. Can a suitable solution be found by Code Table Committee discussion? If not, proceed to (3)
3. Take the discussion to a monthly Arctos Working Group meeting for final resolution.

## Implementation

Once all of the Approval Checklist is appropriately checked and there are no Rejection comments, or in special circumstances by decree of the Arctos Working Group, the change may be made.

Review everything one last time.  Ensure the How-To has been followed. Ensure all checks have been made by appropriate personnel. 

Make changes as described above. Ensure the URL of this Issue is included in the definition. 

Close this Issue.

**DO NOT** modify Arctos Authorities in any way before all points in this Issue have been fully addressed; data loss may result.

## Special Exemptions

In very specific cases and by prior approval of The Committee, the approval process may be skipped, and implementation requirements may be slightly altered. Please note here if you are proceeding under one of these use cases.

1. Adding an existing term to additional collection types may proceed immediately and without discussion, but doing so may also subject users to future cleanup efforts. If time allows, please review the term and definition as part of this step.
2. The Committee may grant special access on particular tables to particular users. This should be exercised with great caution only after several smooth test cases, and generally limited to "taxonomy-like" data such as International Commission on Stratigraphy terminology.

**Comment by dustymc** [comment]
I will plan on proceeding with this about 2023-03-27 if there are no objections.

I will proceed immediately upon approval of each of the involved collections.

Data:
[temp_mexican_wolf_studbook_number.csv.zip](https://github.com/ArctosDB/arctos/files/10963044/temp_mexican_wolf_studbook_number.csv.zip)

Summary:
|guid_prefix|numrecs|
|-----------|--------|
|DGR:Mamm|104|
|MSB:Mamm|1346|
|MSBObs:Mamm|3|


Users:
@campmlc
@AdrienneRaniszewski
@jldunnum


See also https://github.com/ArctosDB/arctos/issues/5771

**Comment by campmlc** [comment]
This cannot proceed for the same reason as #5919. Any endangered species studbook number must remain in the code table as a distinct ID. This will be the case also for bird bands. Anything that has potential use as a linking identifier for mark recapture, captive breeding programs, or other multiple or linked records cannot be merged at this time. 
These identifiers were not in the google doc we discussed at the meeting. We need to pause this process until everyone is on the same page.

**Comment by campmlc** [comment]
This cannot proceed for the same reason as https://github.com/ArctosDB/arctos/issues/5919. Any endangered species studbook number must remain in the code table as a distinct ID. This will be the case also for bird bands. Anything that has potential use as a linking identifier for mark recapture, captive breeding programs, or other multiple or linked records cannot be merged at this time.

**Comment by Jegelewicz** [comment]
> Any endangered species studbook number must remain in the code table as a distinct ID. This will be the case also for bird bands.

I disagree 100% - the change is going to make this so much clearer and I am not sure how to reassure everyone! I have given multiple examples across many issues and I hope that it becomes clear that this change is GOOD.

How about I add the proposed new identifier to all of these records so that you can see how it works? I'll leave the old ones there too so that you can see that the new ones do the same thing the old ones do.

**Comment by Jegelewicz** [comment]
@campmlc These are loading now - check in 20 minutes or so to see how they look

**Comment by AdrienneRaniszewski** [comment]
I'm with Mariel. I don't really understand how this is better. I just looked at one of the records. MSB:Mamm:142631. It now has an "other identifier" 412 assigned by USFWS. But when I look up the wolf in the annual project report it also has a transponder number which is also assigned by USFWS. And in some cases the local id could also be assigned by USFWS. How do I know which is which without there being a more specific name for it than "other identifier"? The issued by field is definitely helpful, but will you be adding new types of identifiers? like a generic "local ID" or "transponder" or "studbook number" so they can be distinguished? It feels like we're losing some info.

**Comment by Jegelewicz** [comment]
> like a generic "local ID" or "transponder" or "studbook number" so they can be distinguished? It feels like we're losing some info.

The issued by for the wolves is [United States Fish and Wildlife Service Mexican Wolf Recovery Program](https://arctos.database.museum/agent/1012059) which should be distinct from other USFWS programs issuing identifiers?

If that is not the case, then perhaps adding a generic "studbook number" might be a good idea?

**Comment by Jegelewicz** [comment]
Also, we can always make the agent more specific - maybe we need a division of [United States Fish and Wildlife Service Mexican Wolf Recovery Program](https://arctos.database.museum/agent/1012059) called United States Fish and Wildlife Service Mexican Wolf Recovery Program Studbook?

**Comment by dustymc** [comment]
>also assigned by USFWS

That should probably be something we always strive to avoid. "Some one of the thousands of employees in this huge agency for some unknown purpose" ain't great.

>distinguished

Better agents is I think the key. Huge Agency-->Some Vague Subdivision ---> Some Specific Branch fits fine in relationships, all can carry addresses and relationships to other agents and etc. From that, I might be able to find someone who has data!

So agreed on the big picture - we should be as specific as possible - but I think agents are just a better mechanism on all counts. (Identical functionality if everyone does the bare minimum correctly, but a lot easier to do correctly and a lot easier to go beyond the bare minimum.)

And I should have known better/looked before typing, because that's precisely what @Jegelewicz did

<img width="1184" alt="Screenshot 2023-03-14 at 9 14 29 AM" src="https://user-images.githubusercontent.com/5720791/225069049-3bd45bd4-f3db-459c-973f-691c9c575d8a.png">

goes to https://arctos.database.museum/agent/1012059 which goes to https://arctos.database.museum/agent/1011108 and https://arctos.database.museum/agent/3679 - these are now much better integrated.

>need a division of [United States Fish and Wildlife Service Mexican Wolf Recovery Program](https://arctos.database.museum/agent/1012059) called United States Fish and Wildlife Service Mexican Wolf Recovery Program Studbook?

If the former issues a bunch of kinds of identifiers and the latter only one, yea, probably.

**Comment by campmlc** [comment]
The studbook number is an ID assigned by the recovery program. It is not an agent. It is critical as it is the unique identifier used across all breeding programs to track a single animal over its lifetime, as well as to track its relationships with other individuals. For these types of IDs - the specific information about the type of ID that we currently capture will be lost if it is converted to other ID. 
And then there is the entity problem. Using other identifier breaks entities as they are currently set up. If I click on the Mexican Wolf Studbook number in the current system, I get search results that are unique to that ID, that I can easily use to create an entity. If I click on the new other identifier entry for the same Mexican wolf record, I  search results from multiple unrelated records in other institutions as well - because the "other identifier" is "412".

**Comment by AdrienneRaniszewski** [comment]
All of my above examples were assigned by USFWS Mexican Wolf Recovery Program. Basically it would continue to be helpful if there's a way to tell apart different types of numbers assigned by the same agent. creating a new agent instead of an identifier type seems weird and counter-intuitive. for example United States Fish and Wildlife Service Mexican Wolf Recovery Program Studbook is not an agent. it's an identifier.

**Comment by campmlc** [comment]
Yes, to Adrienne's point - we need to exclude these types of identifiers from the bulk update until these issues can be resolved. I have flagged potential problems for MSB in the google sheet. But other collections need to review as well. 
If we limit our current cleanup to just ambiguous things like "United States Fish and Wildlife Service" - then we can make progress. But these specific identifiers from endangered species recovery and mark recapture programs, as well as other types of IDs that broadly link acorss multiple collections - need to be managed separately.

**Comment by campmlc** [comment]
And in the meantime, absolutely go ahead with adding the issued by metadata. But leave the ID type.

**Comment by Jegelewicz** [comment]
A demonstration.

## The "old" method for Mexican Wolf identifiers

![image](https://user-images.githubusercontent.com/5725767/225098717-943504d5-df75-4f98-964a-e2a4a6312726.png)

results

![image](https://user-images.githubusercontent.com/5725767/225098826-5f6ddf67-7b27-4ba6-b931-0c88353d76b5.png)

## The "new method

![image](https://user-images.githubusercontent.com/5725767/225100745-4512a459-b8af-480b-8fc7-bc5a026875b6.png)


results

![image](https://user-images.githubusercontent.com/5725767/225100895-e952aa47-ecbd-47d8-831f-ac33984f6f93.png)


Why the difference? I do not have write access for MSBObs:Mamm so those three records did not get a new version of the identifier added.

Nothing is changing except how you enter and search for other identifiers. And if any of these identifiers should be "transponder number which is also assigned by USFWS. And in some cases the local id could also be assigned by USFWS", then they are not correct NOW.

**Comment by Jegelewicz** [comment]
OK now the results are the same - I missed a record when I added the identifiers.

**Comment by campmlc** [comment]
Search from ID table and Entity problem still exists. Happy to do a demo and discussion to see if we can resolve. But until that is resolved, no go on converting these.

**Comment by Jegelewicz** [comment]
> The studbook number is an ID assigned by the recovery program. It is not an agent. It is critical as it is the unique identifier used across all breeding programs to track a single animal over its lifetime, as well as to track its relationships with other individuals. For these types of IDs - the specific information about the type of ID that we currently capture will be lost if it is converted to other ID. And then there is the entity problem. Using other identifier breaks entities as they are currently set up. If I click on the Mexican Wolf Studbook number in the current system, I get search results that are unique to that ID, that I can easily use to create an entity. If I click on the new other identifier entry for the same Mexican wolf record, I search results from multiple unrelated records in other institutions as well - because the "other identifier" is "412".

@campmlc can you please file an issue that details exactly what you need for entities - it is buried here and will not get proper attention. Thanks!

**Comment by campmlc** [comment]
Yes, will do - but have to break for awhile to deal with urgent work
issues. Just need this other identifier thing on hold for the ID types I
request until we can have a broader discussion after this week.

**Comment by AdrienneRaniszewski** [comment]
Agree! Jon will be back next week. He needs a say in all these changes too.
________________________________

**Comment by campmlc** [comment]
Not OK to proceed until resolution of https://github.com/ArctosDB/arctos/issues/6004 and https://github.com/ArctosDB/arctos/issues/6002

**Comment by campmlc** [comment]
#5936 request for studbook number as an ID value.

**Comment by jldunnum** [comment]
I support this request. Having this as an identifier also makes providing information or search instructions to USFWS personnel very unfamiliar with our database much more straighforward.

______________________________________________________________

Jonathan L. Dunnum Ph.D. (he, him, his)
Senior Collection Manager
Division of Mammals, Museum of Southwestern Biology
University of New Mexico
Albuquerque, NM 87131
(505) 277-9262
Fax (505) 277-1351

Chair, Systematic Collections Committee, American Society of Mammalogists
Latin American Fellowship Committee, ASM

MSB Mammals website: http://www.msb.unm.edu/mammals/index.html
Facebook: http://www.facebook.com/MSBDivisionofMammals

Shipping Address:
Museum of Southwestern Biology
Division of Mammals
University of New Mexico
CERIA Bldg 83, Room 204
Albuquerque, NM 87131
________________________________

**Comment by dustymc** [comment]
https://github.com/ArctosDB/arctos/issues/6252 has been completed, https://arctos.database.museum/info/ctDocumentation.cfm?table=CTCOLL_OTHER_ID_TYPE#studbook exists, this can only serve as a denormalizer/point of confusion at this point. @mkoo check please!

**Comment by mkoo** [comment]
I see your point @dustymc but it does satisfy the need to distinguish MWRP ID types (if that is how they want to handle it).

Obviously MVZ decided to create agent profiles for each of our legacy catalogs.

If we go with "studbook" and "eartag" as different IDTypes for a single agent, it would be good to differentiate on the profile page. Example:
![Firefox_Screenshot_2023-06-22T20-18-05 497Z](https://github.com/ArctosDB/arctos/assets/2523089/c0dcdec5-9bf6-4efe-9f22-bce70a095449)

Let me know to file a new issue if you think feasible...

**Comment by dustymc** [comment]
>"studbook" and "eartag" as different IDTypes

I don't want to get too distracted from this, but that just about exactly describes the intent of this.

1. Some identifiable entity (a division of USFWS, perhaps) issues some identifier.
2. That identifier is printed on an eartag, a tattoo, a collar, a PIT tag, ......


Finding some number on some device tells you near-nothing. Knowing about the device tells you nothing. Knowing where the number came from tells you everything.

I'm still not entirely convinced that https://github.com/ArctosDB/arctos/issues/6252 should have happened, but it did and we can deal with it if it starts to look like a bad idea at some future time. For now, I think we just need another checkbox ^^ up there, which will get us one step closer to a normalized state from which everyone should be able to see everything with a lot more clarity. I don't think more issues are necessary.

**Comment by campmlc** [comment]
Different "devices" issued by the same agent/agency carry different number series. It makes a difference to human beings managing samples and data to know whether "12345" is an eartag number vs a studbook number. Forcing the value to be "eartag 12345" will inevitable result in "eartag12345" , "ET12345", "12345", EAR12345" and every other possible variant, making it impossible to search.  I remain opposed to calling eartag numbers and studbook numbers "agents" until the broader community truly understands this fairly radical change in concept and can weigh in - and that is not currently the case.
But since we now have "studbook" as an ID type, which can be "Issued by" agent = USFWS Mexican Wolf Recovery Program, I am ok with proceeding with this. I have edited the original proposal so that the new ID type is "studbook" instead of other identifier.

**Comment by campmlc** [comment]
Note that these records now have two identifiers that will need to be changed. Each record has both "Mexican wolf studbook number" and also "other identifier" issued by USFWS Mexican Wolf Recovery Program. One or the other needs to be deleted during the changeover.

**Comment by Jegelewicz** [comment]
I would like to clarify a few things.

I never meant to propose that we have an agent = "Mexican Wolf Studbook Number". My proposal was to have an agent = "Mexican Wolf Studbook". A studbook "issues" identifiers. I would also expect "Mexican Wolf Studbook" to be a division of [United States Fish and Wildlife Service Mexican Wolf Recovery Program](https://arctos.database.museum/agent/1012059). 

Then, if the [United States Fish and Wildlife Service Mexican Wolf Recovery Program](https://arctos.database.museum/agent/1012059) also issues ear tags, a new agent would be created, say "Mexican Wolf Ear Tag Book" or something like that, which also "issues" some kind of identifier.

In this way, the "kind" of identifier is bound up with the agent.

**Comment by dustymc** [comment]
>"eartag12345" , "ET12345", "12345", EAR12345"

NOTHING will compensate for arbitrarily changing identifiers!

>now have two identifiers 
> am ok with proceeding 

@Jegelewicz can you un-do your addition? I'd like to get this done ASAP and don't want to create a bigger mess in the process.

**Comment by Jegelewicz** [comment]
> @Jegelewicz can you un-do your addition?

Done

**Comment by campmlc** [comment]
Re:
>  My proposal was to have an agent = "Mexican Wolf Studbook". A studbook "issues" identifiers. I would also expect "Mexican Wolf Studbook" to be a division of [United States Fish and Wildlife Service Mexican Wolf Recovery Program](https://arctos.database.museum/agent/1012059).
> 
> Then, if the [United States Fish and Wildlife Service Mexican Wolf Recovery Program](https://arctos.database.museum/agent/1012059) also issues ear tags, a new agent would be created, say "Mexican Wolf Ear Tag Book" or something like that, which also "issues" some kind of identifier.
> 
> In this way, the "kind" of identifier is bound up with the agent.

The problem with this approach is that there is likely no "book" that exists for eartags etc - half the time it is an excel spreadsheet. The eartags are issued by and maintained by an agent, but they themselves are not agents. If we were to go with this approach, we'd also need to create agents not only for every person but every person's collector catalog and preparator catalog, e.g. "Andrew B. Johnson Preparator Catalog" would be an agent. NK numbers would be issued by the "NK Book" agent - and that is not accurate. I think pursuing this approach requires that the community truly understand the scope of what is being proposed, and I do not believe that is the case.

**Comment by campmlc** [comment]
And I don't understand the comment below - please clarify?

> > "eartag12345" , "ET12345", "12345", EAR12345"
> 
> NOTHING will compensate for arbitrarily changing identifiers!

**Comment by mkoo** [comment]
> The problem with this approach is that there is likely no "book" that exists for eartags etc - half the time it is an excel spreadsheet. The eartags are issued by and maintained by an agent, but they themselves are not agents. If we were to go with this approach, we'd also need to create agents not only for every person but every person's collector catalog and preparator catalog, e.g. "Andrew B. Johnson Preparator Catalog" would be an agent. NK numbers would be issued by the "NK Book" agent - and that is not accurate. I think pursuing this approach requires that the community truly understand the scope of what is being proposed, and I do not believe that is the case.

Ah, yes Mariel, you're showing why it's better to create profiles! Now you can leave your institutional knowledge in the public or curatorial (private) box -- otherwise it's only rests with you and the current staff! 

If you dont like the terms 'book', change it! It's just a suggestion from our objective outside perspective. 
[United States Fish and Wildlife Service Mexican Wolf Recovery Program](https://arctos.database.museum/agent/1012059) with different idtypes (eartags vs studbook) or separate agents of "United States Fish and Wildlife Service Mexican Wolf Recovery Program Ear tags" or whatever makes sense. I think there's a lot more flexibility here with the main caveat is that idtypes should be Arctos collections and descriptive not specific.

As we pointed out MVZ went with what seemed simpler to have separate profiles for all our legacy catalogs, a few are actual ledgers, some are lab notebooks, others are who knows what original format, but all need to be tracked and now we can add our tidbits of knowledge for the next hapless generation of curators!

Happy to have a call to discuss too

**Comment by mkoo** [comment]
> Note that these records now have two identifiers that will need to be changed. Each record has both "Mexican wolf studbook number" and also "other identifier" issued by USFWS Mexican Wolf Recovery Program. One or the other needs to be deleted during the changeover.

Yes that will be cleaned up-- the redundancy was to illustrate its usage so you could compare
